packages/horizon/contracts/interfaces/IRecurringCollector.sol

```diff
@@ -0,0 +1,366 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.27;
+
+import { IPaymentsCollector } from "./IPaymentsCollector.sol";
+import { IGraphPayments } from "./IGraphPayments.sol";
+import { IAuthorizable } from "./IAuthorizable.sol";
+
+/**
+ * @title Interface for the {RecurringCollector} contract
+ * @dev Implements the {IPaymentCollector} interface as defined by the Graph
+ * Horizon payments protocol.
+ * @notice Implements a payments collector contract that can be used to collect
+ * recurrent payments.
+ */
+interface IRecurringCollector is IAuthorizable, IPaymentsCollector {
+    // @notice The state of an agreement
+    enum AgreementState {
+        NotAccepted,
+        Accepted,
+        CanceledByServiceProvider,
+        CanceledByPayer
+    }
+
+    // @notice The party that can cancel an agreement
+    enum CancelAgreementBy {
```

Comment:
We probably want to add another party here to represent agreements that got canceled by a third party when force closing a stale allocation.

Reply:
hmm interesting edge case... though I would consider that as cancelled by the indexer?